tests/unit/utils.py

```diff
@@ -239,13 +239,13 @@ def inner(_inst, abs_path):
             if os.path.basename(abs_path) == "{}.yaml".format(parent_dir):
                 log.debug("files loaded so far=%s",
                           _inst.stats_num_files_loaded)
-                assert _inst.stats_num_files_loaded < 2
+                #assert _inst.stats_num_files_loaded < 2
```

Comment:
@dosaboy The scenario tests are throwing exceptions on these assertions. What could be the reason?
tox -e 
```
> tox -e py310 -- TestOpenstackScenarioChecks.test_bug

py310 installed: astroid==2.5.6,attrs==22.1.0,autopage==0.5.1,bashate==2.1.1,cffi==1.15.1,click==8.1.3,cliff==4.0.0,cmd2==2.4.2,cryptography==3.4.8,extras==1.0.0,fasteners==0.18,fixtures==4.0.1,flake8==4.0.1,future==0.18.2,importlib-metadata==5.0.0,isort==5.10.1,lazy-object-proxy==1.8.0,mccabe==0.6.1,mock==4.0.3,pbr==5.11.0,prettytable==3.5.0,progress==1.6,pycodestyle==2.8.0,pycparser==2.21,pyflakes==2.4.0,pylint==2.8.3,pyperclip==1.8.2,python-subunit==1.4.1,PyYAML==6.0,rpdb==0.1.6,simplejson==3.17.6,stestr==4.0.1,stevedore==4.1.0,structr==0.1,testtools==2.5.0,toml==0.10.2,voluptuous==0.13.1,wcwidth==0.2.5,wrapt==1.12.1,zipp==3.10.0
py310 run-test-pre: PYTHONHASHSEED='0'
py310 run-test: commands[0] | stestr run --serial --test-path /workspace/github/hotsos/tests/unit/ TestOpenstackScenarioChecks.test_bug
plugin 'None' cache root not setup
global cache root not setup
plugin 'None' cache root not setup
global cache root not setup
{0} tests.unit.test_openstack.TestOpenstackScenarioChecks.test_bugs [0.013260s] ... FAILED

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    Traceback (most recent call last):

      File "/workspace/github/hotsos/tests/unit/utils.py", line 292, in create_files_inner2
    ret = f(*args, **kwargs)

      File "/workspace/github/hotsos/.tox/py310/lib/python3.10/site-packages/mock/mock.py", line 1346, in patched
    return func(*newargs, **newkeywargs)

      File "/workspace/github/hotsos/tests/unit/utils.py", line 145, in inner
    YScenarioChecker()()

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 117, in __call__
    return self.run_checks()

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 107, in run_checks
    self.load()

      File "/workspace/github/hotsos/hotsos/core/ycheck/scenarios.py", line 34, in load
    plugin_content = YDefsLoader('scenarios').plugin_defs

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 69, in plugin_defs
    loaded = self._get_defs_recursive(path)

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 31, in _get_defs_recursive
    subdefs = self._get_defs_recursive(abs_path)

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 34, in _get_defs_recursive
    if not self._is_def(abs_path):

      File "/workspace/github/hotsos/tests/unit/utils.py", line 249, in inner
    assert _inst.stats_num_files_loaded < 2

    AssertionError

==============================
Failed 1 tests - output below:
==============================

tests.unit.test_openstack.TestOpenstackScenarioChecks.test_bugs
---------------------------------------------------------------

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    Traceback (most recent call last):

      File "/workspace/github/hotsos/tests/unit/utils.py", line 292, in create_files_inner2
    ret = f(*args, **kwargs)

      File "/workspace/github/hotsos/.tox/py310/lib/python3.10/site-packages/mock/mock.py", line 1346, in patched
    return func(*newargs, **newkeywargs)

      File "/workspace/github/hotsos/tests/unit/utils.py", line 145, in inner
    YScenarioChecker()()

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 117, in __call__
    return self.run_checks()

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 107, in run_checks
    self.load()

      File "/workspace/github/hotsos/hotsos/core/ycheck/scenarios.py", line 34, in load
    plugin_content = YDefsLoader('scenarios').plugin_defs

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 69, in plugin_defs
    loaded = self._get_defs_recursive(path)

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 31, in _get_defs_recursive
    subdefs = self._get_defs_recursive(abs_path)

      File "/workspace/github/hotsos/hotsos/core/ycheck/engine/common.py", line 34, in _get_defs_recursive
    if not self._is_def(abs_path):

      File "/workspace/github/hotsos/tests/unit/utils.py", line 249, in inner
    assert _inst.stats_num_files_loaded < 2

    AssertionError



======
Totals
======
Ran: 1 tests in 0.0133 sec.
 - Passed: 0
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 1
Sum of execute time for each test: 0.0133 sec.

==============
Worker Balance
==============
 - Worker 0 (1 tests) => 0:00:00.013260

No tests were successful during the run
ERROR: InvocationError for command /workspace/github/hotsos/.tox/py310/bin/stestr run --serial --test-path tests/unit TestOpenstackScenarioChecks.test_bug (exited with code 1)
```

Reply:
Ok so, this is on me and I need to fix it but, the reason why it is happening is because when you created a test called bugs.yaml, the code searches for bugs.yaml under defs/scenarios/openstack and finds 5 with the same name:

$ find defs/scenarios/openstack/ -name bugs.yaml
defs/scenarios/openstack/octavia/bugs.yaml
defs/scenarios/openstack/keystone/bugs.yaml
defs/scenarios/openstack/barbican/bugs.yaml
defs/scenarios/openstack/neutron/bugs.yaml
defs/scenarios/openstack/nova/bugs.yaml

We used to resolve this by respecting the full namespace i.e. for you it would be keystone/bugs.yaml. So I just need to fix that cause I didn't add it to the new loader. I'll let you know when that is done.

Reply:
this is now fixed on main so try rebasing, you can also remove the comments on assertions and the ```target-name: bugs.yaml``` is no longer necessary (its a noop now really)